lib/PuppeteerSharp/Cdp/ChromeTargetManager.cs

```diff
@@ -101,8 +103,11 @@ private void StoreExistingTargetsForInit()
                     null,
                     _browser.ScreenshotTaskQueue);
 
-                if ((_targetFilterFunc == null || _targetFilterFunc(targetForFilter)) &&
-                    kv.Value.Type != TargetType.Browser)
+                // Targets from extensions and the browser that will not be
+                // auto-attached. Therefore, we should not add them to
+                // #targetsIdsForInit.
+                var skipTarget = kv.Value.Type == TargetType.Browser || kv.Value.Url.StartsWith("chrome-extension://", StringComparison.InvariantCultureIgnoreCase);
+                if ((_targetFilterFunc == null || _targetFilterFunc(targetForFilter)) && !skipTarget)
```

Comment:
Can we check `skipTarget` first or must `_targetFilterFunc` always be run?
E.g. if it has side effects.

Reply:
Good one. `_targetFilterFunc` is not that expensive. But it's true that we already solved `skipTarget`.